.github/workflows/check-query-files-and-compilation.yml

```diff
@@ -1,37 +1,83 @@
 name: Parser compilation and query file check
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    branches:
+      - 'master'
+
+# Cancel any in-progress CI runs for a PR if it is updated
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
 
 jobs:
-  check_compilation_unix_like:
+  check_compilation_unix_latest:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        cc: [ gcc ]
+        nvim_tag: [ nightly ]
+
+    name: Parser compilation
+    runs-on: ${{ matrix.os }}
+    env:
+      CC: ${{ matrix.cc }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ilammy/msvc-dev-cmd@v1.5.0
+      - uses: actions/setup-node@v2
+
+      - name: Install and prepare Neovim
+        env:
+          NVIM_TAG: ${{ matrix.nvim_tag }}
+          TREE_SITTER_CLI_TAG: v0.20.2
+        run: |
+          bash ./scripts/ci-install-${{ matrix.os }}.sh
+
+      - name: Compile parsers Unix like
+        run: |
+          nvim --headless -c "TSInstallSync all" -c "q"
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: parsers-${{ matrix.os }}-${{ matrix.cc }}-x86_64
+          path: parser/*
+
+      - name: Check query files (Unix)
+        if: matrix.os != 'windows-2022'
+        run: nvim --headless -c "luafile ./scripts/check-queries.lua" -c "q"
+
+  check_compilation_all:
```

Comment:
I'm a bit worried about this file getting too complicated by this copy and paste. I'd rather just kick out configuration and cache parser installation

Reply:
I think it's still fine; Neovim does the same. (If there's a better way of skipping workflows inside a matrix job, I'm all ears, though.)

Reply:
(I thought about simplifying the draft job by hardcoding everything, but that seemed even more error-prone.)